airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/task/implementor/ProcessRecordsTask.kt

```diff
@@ -42,49 +46,61 @@ class DefaultProcessRecordsTask(
-                                .lineSequence()
-                                .map {
-                                    when (val message = deserializer.deserialize(it)) {
-                                        is DestinationStreamAffinedMessage -> message
-                                        else ->
-                                            throw IllegalStateException(
-                                                "Expected record message, got ${message::class}"
-                                            )
-                                    }
-                                }
-                                .takeWhile {
-                                    it !is DestinationRecordStreamComplete &&
-                                        it !is DestinationRecordStreamIncomplete
-                                }
-                                .map { it as DestinationRecord }
-                                .iterator()
-                        val batch = streamLoader.processRecords(records, file.totalSizeBytes)
-                        log.info { "Finished processing $file" }
-                        batch
+        registration.use {
+            inputQueue.consume().collect { (streamDescriptor, file) ->
+                log.info { "Fetching stream loader for $streamDescriptor" }
+                val streamLoader = syncManager.getOrAwaitStreamLoader(streamDescriptor)
+                log.info { "Processing records from $file for stream $streamDescriptor" }
+                val batch =
+                    try {
+                        file.localFile.inputStream().use { inputStream ->
+                            val records = inputStream.toRecordIterator()
+                            val batch = streamLoader.processRecords(records, file.totalSizeBytes)
+                            log.info { "Finished processing $file" }
+                            batch
+                        }
+                    } finally {
+                        log.info { "Processing completed, deleting $file" }
+                        file.localFile.toFile().delete()
+                        diskManager.release(file.totalSizeBytes)
                     }
-                } finally {
-                    log.info { "Processing completed, deleting $file" }
-                    file.localFile.toFile().delete()
-                    diskManager.release(file.totalSizeBytes)
-                }
 
-            val wrapped = BatchEnvelope(batch, file.indexRange)
-            log.info { "Updating batch $wrapped for $streamDescriptor" }
-            taskLauncher.handleNewBatch(streamDescriptor, wrapped)
+                val wrapped = BatchEnvelope(batch, file.indexRange, streamDescriptor)
+                log.info { "Updating batch $wrapped for $streamDescriptor" }
+                taskLauncher.handleNewBatch(streamDescriptor, wrapped)
+                if (batch.state != Batch.State.COMPLETE && batch.groupId == null) {
```

Comment:
Q: group id is for when the batch is actually a part of a larger batch so we can join them back together?

Reply:
Yep